src/ride/track.c

```diff
@@ -4880,15 +4881,310 @@ void game_command_remove_track(int *eax, int *ebx, int *ecx, int *edx, int *esi,
+
+    return loc_6CDDFD();
+}
+
+money32 loc_6CDDA0() {
+    // pop     esi
+    // pop     edx
+    return loc_6CDDA2();
+}
+
+
+money32 loc_6CDB35(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp) {
+
+}
+
+money32 loc_6CDAE8(int eax, int ebx, int ecx, int edx, int esi, int edi, int ebp) {
+    // call    map_invalidate_tile_full
+    map_invalidate_tile_full(); // TODO: Arguments
+
+    // movzx   ebp, byte ptr [esp+14h+var_10]
+    ebp = edx;
+
+    // imul    ebp, 260h
+    ebp = ebp * 0x260;
+
+    // inc     ss:word_1362A56[ebp]
+    RCT2_GLOBAL(0x1362A56, uint16) += 1;
+
+    // mov     bl, [esi+2]
+    int bl = RCT2_GLOBAL(esi + 2, uint8);
+
+    // mov     ss:byte_1362952[ebp], bl
+    RCT2_GLOBAL(0x1362952, something)[ebp] = bl;
+
+    // mov     ss:word_136294A[ebp], 0
+    RCT2_GLOBAL(0x136294A, something)[ebp] = 0;
+
+    // cmp     byte ptr [esp+14h+var_8+1], 4
+    int cmp = ebx & 0x00FFFFFF - 4;
+
+    // jnz     short loc_6CDB35
+    if (cmp != 0) {
+        return loc_6CDB35();
+    }
+
+    // test    byte ptr [esp+14h+var_8], 40h
+    int test = ebx - 0x40;
+    // jnz     short loc_6CDB35
+    if (test != 0) {
+        return loc_6CDB35();
+    }
+
+    // push    eax
+    // push    ecx
+
+    // shl     cx, 3
+    // shr     ax, 5
+    // or      ax, cx
+    // mov     ss:word_1362948[ebp], ax
+    RCT2_GLOBAL(0x1362948, uint16) = (uint16) (((eax & 0xFFFF) >> 5) & ((ecx << 3) & 0xFFFF));
+
+    // pop     ecx
+    // pop     eax
+
+    return loc_6CDB35();
+}
+
+money32 do_game_command_set_maze_track(uint16 x, uint8 flags, uint8 direction, uint16 y, uint8 rideIndex, uint8 mode, uint16 unknown, uint16 z) {
+
+	// 0x006CD8CE: game_command_set_maze_track
+	RCT2_GLOBAL(RCT2_ADDRESS_NEXT_EXPENDITURE_TYPE, uint8) = 0;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_X, uint16) = x + 8;
+	RCT2_GLOBAL(RCT2_ADDRESS_COMMAND_MAP_Y, uint16) = y + 8;
+
+	RCT2_GLOBAL(0xF4413E, money32) = 0;
+
+	if (!sub_68B044()) {
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if (!(flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) && RCT2_GLOBAL(RCT2_ADDRESS_GAME_PAUSED, uint8) != 0) {
+		// loc_6CDE2D:
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = STR_CONSTRUCTION_NOT_POSSIBLE_WHILE_GAME_IS_PAUSED;
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	// loc_6CD920:
+	if ((z & 0xF) != 0) {
+		// loc_6CDE38:
+		RCT2_GLOBAL(RCT2_ADDRESS_GAME_COMMAND_ERROR_TEXT, unsigned short) = 954;
+		// loc_6CDE51:
+		return MONEY32_UNDEFINED;
+	}
+
+	if ((flags & GAME_COMMAND_FLAG_APPLY) != 0) {
+		if ((flags & GAME_COMMAND_FLAG_ALLOW_DURING_PAUSED) == 0) {
+			footpath_remove_litter(x, y, z);
+			map_remove_walls_at((uint16) floor2(x, 32), (uint32) floor2(y, 32), z, z + 0x20);
+		}
+	}
+
+	// loc_6CD954:
+	if (map_is_location_owned(floor2(x, 32), floor2(y, 32), z) == 0) {
+		// loc_6CDE51
+		return MONEY32_UNDEFINED;
+	}
+
+	rct_map_element *mapElement = map_get_first_element_at(x / 32, y / 32);
```

Comment:
As stated before this and the next loop is looking for the surface element use `map_get_surface_element_at`